### PR TITLE
support icremental for resourcedocumentprocessor and tocdocumentprocessor

### PR DESCRIPTION
**Pull request by ansyral**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocument.cs [hunk 1]
```diff
@@ -99,5 +99,19 @@ private void BuildCore(TocItemViewModel item, FileModel model, IHostService host
...
+        #region ISupportIncrementalBuild Members
+
+        public bool CanIncrementalBuild(FileAndType fileAndType)
+        {
+            return true;
+        }
+
+        public string GetIncrementalContextHash()
+        {
+            return null;
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocument.cs]: quesion: seems currently IncrementalContextHash all set to null? will we implement it later or not?

Comment (herohua) [inline src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocument.cs]: I think `IncrementalContextHash == null` means the step is stateless and can process those changed files only. @ansyral, correct?

Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocument.cs]: agree with @herohua . the hash would be used to check whether the processor/step can be incremental, currently seems no context info would influence whether the processor can be incremental

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**